akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/JdbcOffsetStore.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.jdbc.internal
+
+import java.sql.Connection
+import java.sql.Timestamp
+import java.time.Clock
+import java.time.Instant
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import akka.Done
+import akka.annotation.InternalApi
+import akka.projection.ProjectionId
+import akka.projection.internal.OffsetSerialization.SingleOffset
+import akka.projection.internal.OffsetSerialization.fromStorageRepresentation
+import akka.projection.internal.OffsetSerialization.toStorageRepresentation
+import akka.projection.jdbc.internal.DialectDefaults.InsertIndices
+import akka.projection.jdbc.javadsl.JdbcSession
+import akka.projection.jdbc.javadsl.JdbcSession.tryWithResource
+import akka.projection.jdbc.javadsl.JdbcSession.withConnection
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[akka] class JdbcOffsetStore[S <: JdbcSession](settings: JdbcSettings, jdbcSessionFactory: () => S, clock: Clock)(
+    implicit val executionContext: ExecutionContext) {
```

Comment:
How will we construct the execution context/dispatcher for the offset store?  The other impls just use the actor system's.  Should we create a default dispatcher for the `JdbcOffsetStore` from configuration?

https://doc.akka.io/docs/akka/current/typed/dispatchers.html

Reply:
Yes, I've been working on that today. Will commit soon. 

The dispatcher will come from a configuration and provided by the JdbcSettings to the store.

Slick runs all operation on a dedicated thread pool. I guess the Cassandra as well.